website/static/js/pointers.js

```diff
@@ -13,196 +13,279 @@ var oop = require('js/oop');
-            self.totalPages(0);
-            self.searchAllProjectsSubmitText(SEARCH_ALL_SUBMIT_TEXT);
-            self.searchMyProjectsSubmitText(SEARCH_MY_PROJECTS_SUBMIT_TEXT);
-        }
-    },
-    addTips: function(elements, data) {
-        elements.forEach(function(element) {
-            var titleText = '';
-            if (data.isRegistration) {
-                titleText = 'Registered: ' + data.dateRegistered.local;
-            } else {
-                titleText = 'Created: ' + data.dateCreated.local + '\nModified: ' + data.dateModified.local;
-            }
-            $(element).tooltip({
-                title: titleText
+            requestNodeLinks.fail(function(xhr){
+                self.searchWarningMsg(xhr.responseJSON && xhr.responseJSON.message_long);
+                count -= 1;
+                if (count === 0){
+                    self.results(nodes);
+                    self.currentPage(self.pageToGet());
+                    self.numberOfPages(Math.ceil(response.links.meta.total / response.links.meta.per_page));
+                    self.addNewPaginators();
+                }
+                self.doneSearching();
             });
         });
+        requestNodes.fail(function(xhr){
+            self.searchWarningMsg(xhr.responseJSON && xhr.responseJSON.message_long);
+            self.doneSearching();
+        });
     },
-    add: function(data) {
-        this.selection.push(data);
-        // Hack: Hide and refresh tooltips
-        $('.tooltip').hide();
-        $('.pointer-row').tooltip();
-    },
-    remove: function(data) {
+    add: function(data){
         var self = this;
-        self.selection.splice(
-            self.selection.indexOf(data), 1
+        var type = self.inputType() === 'nodes' ? 'node_links' : 'registration_links';
+        var addUrl = osfHelpers.apiV2Url('nodes/' + nodeId + '/' + type + '/', {});
+        var request = osfHelpers.ajaxJSON(
+            'POST',
+            addUrl,
+            {
+                'isCors': true,
+                'data': {
+                    'data': {
+                        'type': type,
+                        'relationships': {
+                            'nodes': {
+                                'data': {
+                                    'type': self.inputType(),
+                                    'id': data.id
+                                }
+                            }
+                        }
+                    }
+                }
+            }
         );
-        // Hack: Hide and refresh tooltips
-        $('.tooltip').hide();
-        $('.pointer-row').tooltip();
+        request.done(function (response){
+            self.dirty(true);
+            self.selection.push(data);
+        });
+        request.fail(function(xhr){
+            self.searchWarningMsg(xhr.responseJSON && xhr.responseJSON.message_long);
+        });
     },
-    addAll: function() {
+    remove: function(data){
```

Comment:
Additionally, it is still checked in the add link picker after removing it. You probably need to refresh the page between add/remove links. Or at least re-run the search.